seal.go

```diff
@@ -135,6 +155,148 @@ type KeyCreationParams struct {
 	AuthKey *ecdsa.PrivateKey
 }
 
+// SealKeyToExternalTPMStorageKey seals the supplied disk encryption key to the TPM storage key associated with the supplied public
+// tpmKey. This creates an importable sealed key and is suitable in environments that don't have access to the TPM but do have
+// access to the public part of the TPM's storage primary key. The sealed key object and associated metadata that is required
+// during early boot in order to unseal the key again and unlock the associated encrypted volume is written to a file at the path
+// specified by keyPath.
+//
+// The tpmKey argument must correspond to the storage primary key on the target TPM, persisted at the standard handle.
+//
+// This function expects there to be no file at the specified path. If keyPath references a file that already exists, a wrapped
+// *os.PathError error will be returned with an underlying error of syscall.EEXIST. A wrapped *os.PathError error will be returned if
+// the file cannot be created and opened for writing.
+//
+// This function cannot create a sealed key that uses a PCR policy counter. The PCRPolicyCounterHandle field of the params argument
+// must be tpm2.HandleNull.
+//
+// The key will be protected with a PCR policy computed from the PCRProtectionProfile supplied via the PCRProfile field of the params
+// argument.
+//
+// On success, this function returns the private part of the key used for authorizing PCR policy updates with
+// UpdateKeyPCRProtectionPolicy. This key doesn't need to be stored anywhere, and certainly mustn't be stored outside of the encrypted
+// volume protected with this sealed key file. The key is stored encrypted inside this sealed key file and returned from future calls
+// to SealedKeyObject.UnsealFromTPM.
+//
+// The authorization key can also be chosen and provided by setting
+// AuthKey in the params argument.
+func SealKeyToExternalTPMStorageKey(tpmKey *tpm2.Public, key []byte, keyPath string, params *KeyCreationParams) (authKey TPMPolicyAuthKey, err error) {
+	// params is mandatory.
+	if params == nil {
+		return nil, errors.New("no KeyCreationParams provided")
+	}
+
+	// Perform some sanity checks on params.
+	if params.AuthKey != nil && params.AuthKey.Curve != elliptic.P256() {
+		return nil, errors.New("provided AuthKey must be from elliptic.P256, no other curve is supported")
+	}
+
+	if params.PCRPolicyCounterHandle != tpm2.HandleNull {
+		return nil, errors.New("PCRPolicyCounter must be tpm2.HandleNull when creating an importable sealed key")
+	}
+
+	succeeded := false
+
+	// Compute metadata.
+
```

Comment:
there's a bit of code here that is mostly shared up to parameters with the non importable case, maybe a helper could be shared or not

Reply:
I think I could probably create a helper - I didn't do that initially just because it was quicker this way to get something working.

Reply:
my reason to wonder about that is not the code reduction, is mostly to have one place to change if that are needs to be touched (either bug fixes, or refactoring or other changes)